### PR TITLE
Issue/525

### PR DESCRIPTION
**Pull request by yashLadha**
General Static Component for rendering static pages

### CODE REVIEW

**Comment by yashLadha** [comment]
@nworden  Can you please review and suggest changes?

**Comment by nworden** [comment]
I haven't gotten a chance to do a review yet, but just FYI that test failure looks to me like a flake. If it continues failing on later commits, we can worry about it then, but for now I don't think you need to look into it.

File: ui/src/components/GeneralStatic.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from "react";
+import { FormattedMessage } from "react-intl";
+
+/**
+ * General static component for rendering pagesS
+ *
+ * @class StaticComp
+ * @extends {Component}
+ */
+const StaticComp = ({ subtitle, children }) => (
```
Comment (nworden) [inline ui/src/components/GeneralStatic.js]: Maybe call it `StaticPage` instead.

File: ui/src/components/GeneralStatic.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from "react";
+import { FormattedMessage } from "react-intl";
```
Comment (nworden) [inline ui/src/components/GeneralStatic.js]: I don't think you need to import this.

File: ui/src/components/GeneralStatic.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * General static component for rendering pagesS
```
Comment (nworden) [inline ui/src/components/GeneralStatic.js]: Nit: there's an extra 'S' at the end of this line

File: ui/src/components/GeneralStatic.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * @class StaticComp
+ * @extends {Component}
+ */
+const StaticComp = ({ subtitle, children }) => (
+  <div>
+    <div id='repoheader-info'>
+      <p className='mdc-typography--subtitle1'>{subtitle}</p>
+    </div>
+    { children }
```
Comment (nworden) [inline ui/src/components/GeneralStatic.js]: Please wrap the children in a div with a unique class name (`staticpage-contents` might be a good choice). That way we can apply spacing or stuff like that we might need.

Comment (nworden) [reply ui/src/components/GeneralStatic.js]: I think we'll want to include the footer on every page, so please include it after the children (the footer component is in `components/Footer.js`).

File: ui/src/components/GeneralStatic.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import { FormattedMessage } from "react-intl";
+
+/**
+ * General static component for rendering pagesS
+ *
+ * @class StaticComp
+ * @extends {Component}
+ */
+const StaticComp = ({ subtitle, children }) => (
+  <div>
```
Comment (nworden) [inline ui/src/components/GeneralStatic.js]: We'll probably need a class name to style this div. Could you call it `staticpage-wrapper` or something like that?

File: ui/src/components/GeneralStatic.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    <div id='repoheader-info'>
```
Comment (nworden) [inline ui/src/components/GeneralStatic.js]: You can definitely go ahead with this PR without doing the `RepoHeader` refactor if you want, but could you add a to-do for me to do it? I can never remember for sure how to do JSX comments, but I think it'd be something like this:
`{//TODO(nworden): refactor RepoHeader and use it here}`

File: ui/src/components/GeneralStatic.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from "react";
```
Comment (nworden) [inline ui/src/components/GeneralStatic.js]: Nit: for the sake of consistent style, please use single quotes here, and for `./Footer`  on the next line.

**Pull request merged**